### PR TITLE
[BISERVER-12900]-Existing lock file /opt/pentaho/server/biserver-ee/pentaho-solutions/system/jackrabbit/repository/.lock detected

### PR DESCRIPTION
**Pull request by denisprotopopov**
- added shutdowning repository in contextDestroyed of PentahoSystemReadyListener

### CODE REVIEW

**Comment by denisprotopopov** [comment]
@mbatchelor please look at the changes

**Comment by denisprotopopov** [comment]
@pamval @mbatchelor please look at the changes

**Comment by pamval** [comment]
@pentaho-nbaker @pedrofvteixeira, this change looks sane: shutdown the jcr repository when we're shutting down the server. My question is if we should  reference the jcrrepository directly here or if there's a more abstract representation of the repository or interface we should get?

**Comment by denisprotopopov** [comment]
@pentaho-nbaker @pedrofvteixeira Any updates on this?

File: extensions/src/main/java/org/pentaho/platform/web/http/context/PentahoSystemReadyListener.java [hunk 1]
```diff
@@ -32,7 +32,8 @@
 import org.pentaho.platform.engine.core.system.PentahoSessionHolder;
 import org.pentaho.platform.engine.core.system.PentahoSystem;
 import org.pentaho.platform.util.logging.Logger;
-
+import javax.jcr.Repository;
+import org.apache.jackrabbit.core.RepositoryImpl;
```
Comment (mbatchelor) [inline extensions/src/main/java/org/pentaho/platform/web/http/context/PentahoSystemReadyListener.java]: Shouldn't cast to the Impl. jcrRepository implements org.apache.jackrabbit.api.JackrabbitRepository which is OK to cast to.

Comment (denisprotopopov) [inline extensions/src/main/java/org/pentaho/platform/web/http/context/PentahoSystemReadyListener.java]: @mbatchelor done

**Comment by kolinus** [comment]
@pentaho-nbaker @pedrofvteixeira , @mbatchelor  could you please review

**Comment by denisprotopopov** [comment]
@pamval @pentaho-nbaker @pedrofvteixeira @mbatchelor could you please review

File: extensions/src/main/java/org/pentaho/platform/web/http/context/PentahoSystemReadyListener.java [hunk 2]
```diff
@@ -64,6 +65,17 @@ public void contextInitialized( ServletContextEvent servletContextEvent ) {
 
   @Override
   public void contextDestroyed( ServletContextEvent servletContextEvent ) {
+    Repository jcrRepository = PentahoSystem.get( Repository.class, "jcrRepository", null );
+    if ( jcrRepository == null ) {
+      System.err.println( "Cannot obtain JCR repository. Exiting" );
```
Comment (mbatchelor) [inline extensions/src/main/java/org/pentaho/platform/web/http/context/PentahoSystemReadyListener.java]: Sorry I didn't see this sooner - the class already has a Logger - please use log.error here.

File: extensions/src/main/java/org/pentaho/platform/web/http/context/PentahoSystemReadyListener.java [hunk 3]
```diff
@@ -64,6 +65,17 @@ public void contextInitialized( ServletContextEvent servletContextEvent ) {
...
+      return;
+    }
+    if ( !( jcrRepository instanceof JackrabbitRepository ) ) {
+      System.err.println(
```
Comment (mbatchelor) [inline extensions/src/main/java/org/pentaho/platform/web/http/context/PentahoSystemReadyListener.java]: Sorry I didn't see this sooner - the class already has a Logger - please use log.error here.

Comment (denisprotopopov) [inline extensions/src/main/java/org/pentaho/platform/web/http/context/PentahoSystemReadyListener.java]: @mbatchelor done

**Comment by denisprotopopov** [comment]
@mbatchelor could you please review

**Comment by denisprotopopov** [comment]
@mbatchelor could you please review

**Pull request merged**